### PR TITLE
Better dev startup notifications

### PR DESCRIPTION
**Pull request by shtukas**
As part of improving the developer experience, this introduces a change in the visuals of the startup script that use in development ( called with `make dev` ). We move from the rather confusing useless "progress bar" (which then hangs there) to a simpler display that let the developer know when the local dev server is running. 

**Before**

screenshot 2018-12-10 at 16 29 34

screenshot 2018-12-10 at 16 29 35

screenshot 2018-12-10 at 16 29 41

**After**

screenshot 2018-12-10 at 16 30 22

screenshot 2018-12-10 at 16 30 27

### CODE REVIEW

**Comment by shtukas** [comment]
Little detail: My terminal is blue, but that's not common. There isn't a usability problem with the url displayed in blue. It's my fault. I should have better colours :)

File: scripts/webpack/frontend.js [hunk 1]
```diff
@@ -5,13 +5,11 @@ const glob = promisify(require('glob'));
 const { smart: merge } = require('webpack-merge');
 const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
 const ReportBundleSize = require('./plugins/report-bundle-size');
-const Progress = require('./plugins/progress');
```
Comment (SiAdcock) [inline scripts/webpack/frontend.js]: Can we delete this plugin, and maybe some of its dependencies?

**Comment by SiAdcock** [comment]
Okay, so a tiny bit more digging, here's what's happening:

1. The server and browser bundles compile in parallel
2. Their progress is jointly reported by our custom progress reporter
3. When the browser bundle finished compiling, we get the lovely [`FriendlyErrorsWebpackPlugin`](https://github.com/guardian/dotcom-rendering/blob/master/scripts/webpack/browser.js#L7-L15) message (as shown in the second "After" screenshot above
4. However the server bundle has not finished compiling. The progress indicator continues increasing and obliterates our lovely message!
5. Finally when both server and browser bundles are complete, the progress indicator stops

So if we want to still preserve the progress indicator, maybe we can move the `friendlyErrorsWebpackPlugin` to the `frontend` config and show the completion message after all bundles have finished compiling?

**Comment by shtukas** [comment]
@SiAdcock 
Or we could do like @nicl suggests: Simpler is better :)

File: scripts/webpack/frontend.js (hunk 1 above)
Comment (shtukas) [reply scripts/webpack/frontend.js]: Done.

**Comment by SiAdcock** [comment]
It's a bit reductionist, no? From a user experience perspective, do we think it's useful to have a visual indication that something is happening? CLI tools that provide this feel a bit more polished. I guess that's what it comes down to.

**Comment by SiAdcock** [comment]
@GHaberis Good point. To guarantee that [this message](https://github.com/guardian/dotcom-rendering/blob/master/scripts/webpack/browser.js#L7-L15) is displayed when both server and browser bundles have completed, we will have to move it up to the [frontend config plugins](https://github.com/guardian/dotcom-rendering/blob/master/scripts/webpack/frontend.js#L75-L88)

**Comment by shtukas** [comment]
> It's a bit reductionist, no?

Just answering to this. I would support a progress bar, if the process took a couple of minutes, but I just don't see the point of an intrusive progress bar for a process that takes few seconds above all that the user is told "Starting frontend DEV server" :)

Anyway, to move forward wit this. What is the next step to make the PR ready ?

**Comment by SiAdcock** [comment]
Fair enough. I think this is almost ready to go, but it would be great if you could `yarn remove` the following dependencies that are no longer being used:

- `term-size`
- `log-update`

In a separate PR, we can ensure the message appears at the correct time. Or you can try fixing it here, it's up to you 🙂

**Comment by shtukas** [comment]
@sihil 
Dependencies removed. Thanks for pointing that out. For the record we decided that depcheck ( https://stackoverflow.com/questions/22675725/find-unused-npm-packages-in-package-json ) is something we might try, as I had missed them thinking out current `make validate` would highlight them.  

> In a separate PR, we can ensure the message appears at the correct time. 

Separate PR.  But just to clarify this point thought. The server is technically listening before the message appears but is not responsive while the server bundle is being built. Therefore I thought that it made perfect sense that the message appears when it does.

**Comment by shtukas** [comment]
@SiAdcock 
Dependencies removed. Thanks for pointing that out. For the record we decided that depcheck ( https://stackoverflow.com/questions/22675725/find-unused-npm-packages-in-package-json ) is something we might try, as I had missed them thinking out current make validate would highlight them.

>In a separate PR, we can ensure the message appears at the correct time.

Separate PR. But just to clarify this point thought. The server is technically listening before the message appears but is not responsive while the server bundle is being built. Therefore I thought that it made perfect sense that the message appears when it does.

**Comment by SiAdcock** [comment]
> The server is technically listening before the message appears but is not responsive while the server bundle is being built

Ha! I'm sure that's true, but from a developer perspective don't you think a non-listening server and a non-responsive server are functionally equivalent? I would prefer to see some sort of a message that told me when the dev server is ready to respond

**Comment by SiAdcock** [comment]
Ah right, I think my mental model was incorrect to begin with. Makes sense, thanks for clarifying 🙂

**Pull request merged**